In [1]:
import pandas as pd
import statsmodels.api as sm

In [2]:
#Loading the data from Excel file
data= pd.read_excel("Dataset_1.xlsx")
data

DESCRIPTION  EXTRACT_DATE  \
0         A Lovely Apartment Perfect As Your Next Home      20211028   
1            A Resort Style Lifestyle With Water Views      20211028   
2         A Splendid Apartment With Multiple Functions      20211028   
3                     AMAZING 2BD DIRECT FROM LANDLORD      20211028   
4          AMAZING VIEW | Book Now! | Flexible Payment      20211028   
..                                                 ...           ...   
353        The Most Classy Apt in Al Reem | Prime Area      20211028   
354  Top Notch 2 BR in World-Class Tower Now Available      20211028   
355                  ZERO Commission | With Maids Room      20211028   
356               Spacious Modern Living 2BR 04 Layout      20211028   
357     Spacious and Clean 2BR with Balcony High Floor      20211028   

    DATA_SOURCE PROPERTY_TYPE        LOCATION     REGION   PRICE  FLOOR_SIZE  \
0      Dubizzle          Rent      Yas Island  Abu Dhabi   85000        1068   
1      Dubizzle          Rent      Yas Island  Abu Dhabi   79960        1050   
2      Dubizzle          Rent      Yas Island  Abu Dhabi   85000        1044   
3      Dubizzle          Rent      Yas Island  Abu Dhabi  115000        1472   
4      Dubizzle          Rent      Yas Island  Abu Dhabi   84995        1674   
..          ...           ...             ...        ...     ...         ...   
353    Dubizzle          Rent  Al Reem Island  Abu Dhabi   80000        1248   
354    Dubizzle          Rent  Al Reem Island  Abu Dhabi   90000        1450   
355    Dubizzle          Rent  Al Reem Island  Abu Dhabi  136000        1858   
356    Dubizzle          Rent  Al Reem Island  Abu Dhabi   87000        1507   
357    Dubizzle          Rent  Al Reem Island  Abu Dhabi   72000        1250   

     NO_OF_BEDROOMS  NO_OF_BATHROOMS  PRICE_SQFT Unnamed: 11  
0                 2                2   79.588015         NaN  
1                 2                2   76.152381         NaN  
2                 2                2   81.417625         NaN  
3                 2                3   78.125000         NaN  
4                 2                3   50.773596         NaN  
..              ...              ...         ...         ...  
353               2                3   64.102564         NaN  
354               2                3   62.068966         NaN  
355               2                4   73.196986         NaN  
356               2                3   57.730591         NaN  
357               2                2   57.600000         NaN  

[358 rows x 12 columns]

In [3]:
# Filter for 2-bedroom houses as specified in the user's end 
filtered_data = data[(data['NO_OF_BEDROOMS'] == 2) & data['FLOOR_SIZE'].notna() &
                     data['NO_OF_BATHROOMS'].notna() & data['PRICE_SQFT'].notna()]

filtered_data.describe(), filtered_data.head()


(       EXTRACT_DATE          PRICE   FLOOR_SIZE  NO_OF_BEDROOMS  \
 count         358.0     358.000000   358.000000           358.0   
 mean     20211028.0   90035.544693  1320.986034             2.0   
 std             0.0   18789.327604   226.662874             0.0   
 min      20211028.0   62500.000000   678.000000             2.0   
 25%      20211028.0   77250.000000  1108.000000             2.0   
 50%      20211028.0   85000.000000  1308.000000             2.0   
 75%      20211028.0   99999.750000  1500.000000             2.0   
 max      20211028.0  184000.000000  2421.000000             2.0   
 
        NO_OF_BATHROOMS  PRICE_SQFT  
 count       358.000000  358.000000  
 mean          2.807263   68.772608  
 std           0.583866   11.571150  
 min           1.000000   45.072115  
 25%           2.000000   60.084823  
 50%           3.000000   66.381818  
 75%           3.000000   76.691992  
 max           5.000000  111.034004  ,
                                     DESCRI

In [4]:
# Defining the independent variables and dependent variable
X = filtered_data[['FLOOR_SIZE', 'NO_OF_BATHROOMS', 'PRICE_SQFT']]
y = filtered_data['PRICE']

In [5]:
# Adding a constant to the model for the intercept

X = sm.add_constant(X)

In [6]:
# Performing the multiple regression analysis
model = sm.OLS(y, X).fit()

In [7]:
# Summary of the regression analysis

model_summary = model.summary()
model_summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  PRICE   R-squared:                       0.985
Model:                            OLS   Adj. R-squared:                  0.985
Method:                 Least Squares   F-statistic:                     7839.
Date:                Tue, 05 Nov 2024   Prob (F-statistic):          2.96e-323
Time:                        09:50:22   Log-Likelihood:                -3276.8
No. Observations:                 358   AIC:                             6562.
Df Residuals:                     354   BIC:                             6577.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const           -9.756e+04   1269.713    -76.835      0.000      -1e+05   -9.51e+04
FLOOR_SIZE         69.6993      0.648    107.591      0.000      68.425      70.973
NO_OF_BATHROOMS   496.9900    245.695      2.023      0.044      13.785     980.195
PRICE_SQFT       1368.6748     11.095    123.357      0.000    1346.854    1390.496
==============================================================================
Omnibus:                      122.851   Durbin-Watson:                   1.888
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              919.750
Skew:                          -1.225   Prob(JB):                    1.90e-200
Kurtosis:                      10.460   Cond. No.                     1.40e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.4e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

R-squared: 0.985, suggesting that approximately 98.5% of the variance in price can be explained by floor size, number of bathrooms, and price per square foot.

FLOOR_SIZE: 𝑝 < 0.001 highly significant.

NO_OF_BATHROOMS: 𝑝 = 0.044 also significant, though less  than the other variables.

PRICE_SQFT: 𝑝 < 0.001 highly significant.

Since all independent variables show significant p-values (less than 0.05), we reject the null hypothesis. This suggests that floor size, number of bathrooms, and price per square foot each have a statistically significant effect on price at a 95% confidence level.